src/main/java/org/radarbase/sink/hdfs/AvroFormatRadar.java

```diff
@@ -28,24 +31,32 @@
 import io.confluent.connect.storage.format.RecordWriterProvider;
 import io.confluent.connect.storage.format.SchemaFileReader;
 import io.confluent.connect.storage.hive.HiveFactory;
+import java.util.Map;
 import org.apache.hadoop.fs.Path;
 
 /**
- * Extended AvroFormat class to support custom AvroRecordWriter to allow writting key and value to
+ * Extended AvroFormat class to support custom AvroRecordWriter to allow writing key and value to
  * HDFS.
  */
 public class AvroFormatRadar implements Format<HdfsSinkConnectorConfig, Path> {
     private final AvroData avroData;
 
+    /**
+     * Avro Format with fixed AvroData instance.
+     * @param storage data storage used for configuration.
+     */
     public AvroFormatRadar(HdfsStorage storage) {
-        AvroDataConfig avroConfig = new AvroDataConfig.Builder()
-                .with(AvroDataConfig.CONNECT_META_DATA_CONFIG, false)
-                .with(AvroDataConfig.SCHEMAS_CACHE_SIZE_CONFIG,
-                        storage.conf().getInt(SCHEMA_CACHE_SIZE_CONFIG))
-                .with(AvroDataConfig.ENHANCED_AVRO_SCHEMA_SUPPORT_CONFIG, true)
-                .build();
-
-        this.avroData = new AvroData(avroConfig);
+        @SuppressWarnings("unchecked")
+        Map<String, Object> conf = (Map<String, Object>) storage.conf().plainValues();
+
+        this.avroData = new AvroData(new AvroDataConfig.Builder()
+                .with(CONNECT_META_DATA_CONFIG,
+                        conf.getOrDefault(CONNECT_META_DATA_CONFIG, false))
```

Comment:
Why are we trying to a get a config option which we know will never be present ?

Reply:
How do we know it will never be present? Can it not be set in the connector config?